plugins/nodes/src/js/pages/nodes/NodeDetailTab.js

```diff
@@ -62,6 +62,22 @@ class NodeDetailTab extends React.Component {
                 {MesosStateStore.get("lastMesosState").version}
               </ConfigurationMapValue>
             </ConfigurationMapRow>
+            <ConfigurationMapRow>
+              <ConfigurationMapLabel>
+                Region
+              </ConfigurationMapLabel>
+              <ConfigurationMapValue>
+                {node.getRegionName() || "\u2014"}
```

Comment:
We should abstract this into a method where returns `node.getRegionName() || "\u2014"` because we are using multiple times

Reply:
We are using it twice, once per file. I would feel uncomfortable to add it to the struct. Also to add a Util for 2 occurrences feels more like an overhead.

I see this type of change more as being a part from the `ConfigurationMapValue` component. To render a `\u2014` if there is no value. How do you feel about me removing the default value. And create an issue to add the default m dash behaviour to the component. I feel it belongs there.

Reply:
We are using it twice, once per file. I would feel uncomfortable to add it to the struct. Also to add a Util for 2 occurrences feels more like an overhead.

I see this type of change more as being a part from the `ConfigurationMapValue` component. To render a `\u2014` if there is no value. How do you feel about me removing the default value. And create an issue to add the default m dash behaviour to the component. I feel it belongs there.

Reply:
In my Framework Configuration PR, I added another prop to `HashMapDisplay` for [emptyValue](https://github.com/dcos/dcos-ui/pull/2442/files#diff-22971d009a414c242d15d6666a1e5155R80), where "\u2014" can be specified.

Reply:
What I meant was to add in the same file not in structs nor a new/or existent util, but this is not a big deal though

Reply:
I do not see any value in that as we would have one function which would be called once in that file. Or we would have a util functions like:
```JS
function getValueOrEmptyValue(value) {
  return value || "\u2014";
}

// ...

<ConfigurationMapValue>
  {getValueOrEmptyValue(node.getRegionName())}
</ConfigurationMapValue>
// ...
<ConfigurationMapValue>
  {getValueOrEmptyValue(node.getZoneName())}
</ConfigurationMapValue>


// Or specific

function getRegionName(node) {
  return node.getRegionName() || "\u2014";
}
function getZoneName(node) {
  return node.getZoneName() || "\u2014";
}

// ...

<ConfigurationMapValue>
  {getRegionName(node)}
</ConfigurationMapValue>
// ...
<ConfigurationMapValue>
  {getZoneName(node)}
</ConfigurationMapValue>

```

I see what you want to change here, but I really think that change is better placed in the `ConfigurationMapValue` component. So that the user does not need to care about the empty value. This would result in

```JS
<ConfigurationMapValue>
  {node.getRegionName()}
</ConfigurationMapValue>
```

And regarding adding the emptyValue prop I would not go that far right now as the requirements right now say that empty values only need to be displayed as "\u2014", but I do like the idea to have it configurable as soon as this is becoming a requirement.

Reply:
makes sense, thanks @Poltergeist